packages/guides/src/Compiler/NodeTransformers/MoveAnchorTransformer.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\NodeTransformers;
+
+use ArrayIterator;
+use LogicException;
+use phpDocumentor\Guides\Compiler\CompilerContext;
+use phpDocumentor\Guides\Compiler\NodeTransformer;
+use phpDocumentor\Guides\Compiler\ShadowTree\TreeNode;
+use phpDocumentor\Guides\Nodes\AnchorNode;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\Nodes\SectionNode;
+
+use function spl_object_hash;
+
+/** @implements NodeTransformer<AnchorNode> */
+final class MoveAnchorTransformer implements NodeTransformer
+{
+    /** @var array<string, string> */
+    private array $seen = [];
```

Comment:
Should this be a WeakMap? So that you don't have to deal with hashes?

Reply:
I never used WeakMap, but it seems to be a very good use case here!